### PR TITLE
feat: exclude operators with slashed validators in the frame

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: src/modules/csm/csm.py [hunk 1]
```diff
@@ -219,6 +220,14 @@ def calculate_distribution(self, blockstamp: ReferenceBlockStamp) -> tuple[int,
                     # is not presented in the aggregates (e.g. exited, pending for activation etc).
                     continue
 
+                if v.validator.slashed:
+                    # It means that validator was active during the frame
+                    # and got slashed and didn't meet the exit epoch,
+                    # so we should not count such operator in the distribution
+                    if distribution.get(no_id):
```
Comment (madlabman) [inline src/modules/csm/csm.py:227]: The check seems to be redundant.

File: src/modules/csm/csm.py [hunk 2]
```diff
@@ -219,6 +220,14 @@ def calculate_distribution(self, blockstamp: ReferenceBlockStamp) -> tuple[int,
                     # is not presented in the aggregates (e.g. exited, pending for activation etc).
                     continue
 
+                if v.validator.slashed:
```
Comment (madlabman) [inline src/modules/csm/csm.py:223]: Move it before try..catch block.

File: src/modules/csm/csm.py (hunk 1 above)
Comment (vgorkavenko) [reply src/modules/csm/csm.py]: removed

File: src/modules/csm/csm.py (hunk 2 above)
Comment (vgorkavenko) [reply src/modules/csm/csm.py]: we can't do that. otherwise we will remove good operators that had slashed validator in the previous frame, for example

**Review by madlabman — APPROVED** [review]

**Review by F4ever — APPROVED** [review]

**Pull request merged**